ql/src/java/org/apache/hadoop/hive/ql/optimizer/physical/Vectorizer.java

```diff
@@ -2594,7 +2594,9 @@ private boolean validateMapJoinDesc(MapJoinDesc desc) {
       return false;
     }
     List<ExprNodeDesc> keyExprs = desc.getKeys().get(posBigTable);
-    if (!validateExprNodeDesc(keyExprs, "Key")) {
+    if (!validateExprNodeDescNoComplex(keyExprs, "Key")) {
```

Comment:
A few lines up (2592-2593) there is a method that seems to control if complex types are allowed or not. How do we choose which one to use?

Reply:
The above lines checks for filter expression. The issue is with the join keys.

Reply:
OK thanks for clarifying.